Fetch events live from Eventful API

liuche opened the pull request:
I ended up in some kind of horrible circular dependency trap, and I didn't want to do a refactor of request_handler along with these changes. I'll file an issue to clean up request_handler though, mainly separating out the database handling, the location calls, and the event calls.

Basically, these all end up being kind of dependent because the request_handler contains all the db writing code, and imported the events.py code for event handling, but the events.py code also handles fetching venues for events (from Pull #45).

This is rebased on top of Pull #45 .

jhugman commented on app/request_handler.py:
```diff
@@ -141,11 +164,90 @@ def searchLocation(lat, lng):
       }
     )
+
+def getEventfulEventObj(event):
```
Definitely getting to the limit of size/responsibility for `request_handler.py`. We're putting down tech debt here to have everything in this file.

jhugman commented on app/request_handler.py:
```diff
@@ -141,11 +164,90 @@ def searchLocation(lat, lng):
+# Fetching events from Google Calendar
+
+DAY_DATETIME = datetime.timedelta(days=1)
+scheduler = sched.scheduler(time.time, time.sleep)
```
Once we're starting to put things in schedulers, we are pretty far away from handling requests.

Suggest putting this into a separate file.

jhugman commented:
I think the best we can do is admit this is tech debt, file an issue to fix it.

The correct course is to separate the firebase read/writing into its own file, then going from there.

liuche commented:
![](https://github.trello.services/images/mini-trello-icon.png) [As a user, I want to be able to see events around me from the eventful service so I can use the app outside of Hawaii.](https://trello.com/c/7FZp1GZG/214-as-a-user-i-want-to-be-able-to-see-events-around-me-from-the-eventful-service-so-i-can-use-the-app-outside-of-hawaii)

liuche commented:
I'm assuming that this would merge after the island-crawler, so this updated pull request is rebased on top of that, and addresses the comments for both events pull requests.

liuche commented:
Cherry-picked these onto Pull #56 .
